scorm.js

```diff
@@ -406,14 +406,31 @@ Scorm.prototype.SetInteraction = function ( interaction_name, objective_name, ou
 
 		this.SetValue('cmi.interactions.'+int_id+'.id', interaction_name, skip_checking);
 		this.SetValue('cmi.interactions.'+int_id+'.objectives.0.id', objective_name, skip_checking);
-		if (this.mode == '2004') 
+		if (this.mode == '2004') {
 			this.SetValue('cmi.interactions.'+int_id+'.description', description);
+            this.SetValue('cmi.interactions.'+int_id+'.type', interaction_type);
+        }
 		// increment
 		this.interactions.push(interaction_name);
 	}
-
+    
 	if (learner_response) {
-		this.SetValue('cmi.interactions.'+int_id+'.student_response', learner_response, skip_checking);
+        if (this.mode == '2004' && interaction_type && correct_response_pattern) {
+            // Set the response to be the id
+            // [:] and ' ' are illegal characters but [.] and [,] are okay
```

Comment:
We're ignoring everything to the right of the question id so it should never have a space